components/clp-package-utils/clp_package_utils/general.py

```
@@ -166,6 +170,48 @@ def get_clp_home():
+    :param method: HTTP method.
+    :param max_attempts: Number of attempts before giving up. Must be >= 1.
+    :param retry_delay: Seconds to wait between retries.
+    :param timeout: Seconds to wait for a response per attempt.
+    :return: The HTTP response.
+    :raise ValueError: If max_attempts < 1.
+    :raise urllib.error.URLError: If all attempts fail due to a network-level error.
+    :raise urllib.error.HTTPError: If all attempts fail with an HTTP error status.
+    :raise OSError: If all attempts fail due to a socket-level error.
+    """
+    if max_attempts < 1:
+        raise ValueError(f"max_attempts must be >= 1, got {max_attempts}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)